core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -57,7 +57,7 @@
 
 /** A communication channel for making outgoing RPCs. */
 @ThreadSafe
-public final class ChannelImpl extends Channel {
+public final class ChannelImpl extends CallFactory implements Channel {
```

Comment:
If `ChannelImpl` needs to extend `CallFactory` too, it's an indication that `Channel` as an interface is not a complete abstraction for its implementations. It would look cleaner if `ChannelImpl` only implemented `Channel`, and you called `ChannelImpl.callFactory()` whenever a `CallFactory` is required.